pages/api/og.tsx

```diff
@@ -43,22 +33,27 @@ export const config = {
 export default function (req: NextRequest) {
   try {
     const { searchParams } = new URL(req.url);
-    const testUrl = new URL(req.url);
-    const hasTitle = searchParams.has('title');
-    const title = hasTitle
-      ? searchParams.get('title')?.slice(0, 100)
-      : 'My Store';
-    const hasColor = searchParams.get('color')!;
-    const color = hasColor ? hasColor : '000000';
-    const foreground =
-      lightOrDark(`#${color}`) == 'light' ? '000000' : 'FFFFFF';
-    const logo = searchParams.get('logo');
+
+    // Defaults if no query parameters are provided
+    const defaultForeColor = '#FFFFFF';
+    const defaultColor = '#000000';
+
+    const title = searchParams.get('title')?.slice(0, 100) || 'My Store';
+
+    // Get the foreground from the search params or use a default
+    const foreground = searchParams.get('foreground') || defaultForeColor;
+
+    // Determine the background color: If not set, compute based on the text color
+    const color = searchParams.get('color') ||
+      (lightOrDark(`#${foreground}`) === 'light' ? defaultForeColor : defaultColor);
+
+    const logoSrc = searchParams.get('logo');
 
     return new ImageResponse(
       (
         <div
           style={{
-            backgroundColor: `#${color}`,
+            backgroundColor: color,
```

Comment:
Looks like color was assumed not to include the leading hash sign, and now we're assuming it does.  Likely not a problem with the usages we know about, but in the spirit of backwards compatibility you should probably prepend the hash if there isn't one.